pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/util/StringSuggestionTypeaheadComparator.java

```diff
@@ -0,0 +1,72 @@
+package org.broadinstitute.ddp.util;
+
+import java.util.Comparator;
+import java.util.regex.Pattern;
+
+/**
+ * Comparator that can be used to sort results for
+ * the picklist option typeahead.  The general approach is favoring leftwards
+ * occurrences, in particular word starts, allowing for a starting
+ * parentheses.
+ */
+public class StringSuggestionTypeaheadComparator implements Comparator<String> {
+
+    private final String searchTerm;
+
+    /**
+     * Create a new one, using the given search term
+     * @param searchTerm a simple search term, provided by the
+     *                   user.  Not a regex.
+     */
+    public StringSuggestionTypeaheadComparator(String searchTerm) {
+        this.searchTerm = searchTerm.toUpperCase();
+    }
+
+    @Override
+    public int compare(String string1, String string2) {
+        String string1Name = string1.toUpperCase();
+        String string2Name = string2.toUpperCase();
+        boolean string1StartsWithSearchTerm = string1Name.startsWith(searchTerm);
+        boolean string2StartsWithSearchTearm = string2Name.startsWith(searchTerm);
+
+        // bias search results so that matches at the start of the name are favored
+        if (string1StartsWithSearchTerm && string2StartsWithSearchTearm) {
+            return string1Name.compareTo(string2Name);
+        } else if (string1StartsWithSearchTerm) {
+            return -1;
+        } else if (string2StartsWithSearchTearm) {
+            return 1;
+        } else {
+            // if the search is not at the start of the term, prefer hits that are at start of words
+            Pattern startOfWordMatch = Pattern.compile("\\b(\\))*" + Pattern.quote(searchTerm));
```

Comment:
I don't think this pattern is doing what the comment says it should be doing
![Screen Shot 2022-03-14 at 5 59 47 PM](https://user-images.githubusercontent.com/29984380/158268328-6a57a982-0be9-4cca-b37e-e0a3bc1e669d.png)
Note that it matches "hello" but not the last example that has the word hello.

Reply:
Its clone of existing in-use DrugTypeaheadComparator and CancerTypeaheadComparator. Planned to refactor to use this to avoid code duplication.
Will check it out any way to see if its missing startOfWord matches

Reply:
> Had no idea you could use the + operator on Pattern! Had to try it out.

Two String's are combined and then converted into Pattern 
Pattern.compile(regexString + searchString);

Reply:
Code seem to work fine as intended.
DrugTypeaheadComparatorTest covers the scenarios that you mentioned Marco